addons/l10n_in_gstin_status/models/res_partner.py

```diff
@@ -0,0 +1,106 @@
+
+    @api.depends('vat')
+    def _compute_l10n_in_gst_status(self):
+        """
+        Reset GST Status Whenever the `vat` of partner changes
+        """
+        for partner in self:
+            if partner.country_code == 'IN':
+                partner.l10n_in_gstin_verified_status = False
+                partner.l10n_in_gstin_verified_date = False
+
+    def _l10n_in_gstin_status_get_url(self, company):
+        if company.sudo().l10n_in_edi_production_env:
+            default_endpoint = DEFAULT_IAP_ENDPOINT
+        else:
+            default_endpoint = DEFAULT_IAP_TEST_ENDPOINT
+        endpoint = self.env["ir.config_parameter"].sudo().get_param("l10n_in_gstin_status.endpoint", default_endpoint)
+        url = "%s%s" % (endpoint, '/iap/l10n_in_reports/1/public/search')
+        return url
+
+    def get_l10n_in_gstin_verified_status(self):
+        self.ensure_one()
+        if not self.vat:
+            raise ValidationError(_("Please enter the GSTIN"))
+        url = self._l10n_in_gstin_status_get_url(self.env.company)
+        params = {
+            "account_token": self.env["iap.account"].get("l10n_in_edi").account_token,
+            "dbuuid": self.env["ir.config_parameter"].sudo().get_param("database.uuid"),
+            "gstin_to_search": self.vat,
+        }
+        try:
+            response = jsonrpc(url, params=params, timeout=25)
+        except AccessError:
+            raise UserError(_("Unable to connect with GST network"))
+        if response.get('error') and any(e.get('code') == 'no-credit' for e in response['error']):
+            return self.env["bus.bus"]._sendone(self.env.user.partner_id, "iap_notification",
+                {
+                    "type": "no_credit",
+                    "title": "Not enough credits to check GSTIN status",
+                    "get_credits_url": self.env["iap.account"].get_credits_url(service_name="l10n_in_edi"),
+                },
+            )
+        gst_status = response.get('data', {}).get('sts', "")
+        if gst_status.casefold() == 'active':
+            l10n_in_gstin_verified_status = True
+        elif gst_status:
+            l10n_in_gstin_verified_status = False
+            date_from = response.get("data", {}).get("cxdt", '')
+            if date_from and re.search(r'\d', date_from):
+                message = _(
+                    "GSTIN %(vat)s is %(status)s and Effective from %(date_from)s",
+                    vat=self.vat,
+                    status=gst_status,
+                    date_from=date_from,
+                )
+            else:
+                message = _(
+                    "GSTIN %(vat)s is %(status)s, effective date is not available.",
+                    vat=self.vat,
+                    status=gst_status
+                )
+            self.message_post(body=message)
+        else:
+            _logger.info("GST status check error %s", response)
+            raise UserError(_("Error in getting GST status. please try again"))
```

Comment:
I have been getting this all the time checking on runbot https://66355233-master-all.runbot164.odoo.com/odoo/customers/147

and it is not very helpful...   It is like: we get an error and we do not know why it is there. (but it could connect)  We should be more careful as to why this happens.  (as this case should not even happen)

Reply:
Checking in the logs of the runbot, we get this: 
2024-08-02 08:35:05,297 29 INFO 66355233-master-all odoo.addons.l10n_in_gstin_status.models.res_partner: GST status check error {'error': [{'code': 'SWEB_9035', 'message': 'Invalid GSTIN / UID'}]} 
See https://runbot164.odoo.com/runbot/static/build/66355233-master/logs/run.txt

We should explicitly say that the GSTIN is invalid.  (and not let the user retry)